#779: Unit tests for Modem classes from core.net

djurod opened the pull request:
- Prepared unit tests for classes ModemInterfaceAddressConfigImpl and
  ModemInterfaceConfigImpl from package org.eclipse.kura.core.net.modem
- Fixed or implemented comparison of classes:
  ModemInterfaceAddressConfigImpl and ModemInterfaceAddressImpl

Signed-off-by: Djuro Drljaca <djuro.drljaca@comtrade.com>

cdealti commented on kura/org.eclipse.kura.core.net/src/main/java/org/eclipse/kura/core/net/modem/ModemInterfaceAddressConfigImpl.java:
```diff
@@ -39,63 +39,36 @@ public void setNetConfigs(List<NetConfig> configs) {
+		if (!super.equals(obj)) {
+			return false;
+		}
+		if (!(obj instanceof ModemInterfaceAddressConfigImpl)) {
```
This could be checked earlier I believe

djurod replied:
As far as I remember this is how Eclipse generates hashCode() and equals(), but I agree that making the "cheaper" check should be done earlier. I'll fix it.

cdealti commented on kura/org.eclipse.kura.core.net/src/main/java/org/eclipse/kura/core/net/modem/ModemInterfaceAddressImpl.java:
```diff
@@ -94,4 +94,50 @@ public void setConnectionType(ModemConnectionType connectionType) {
+		if (!(obj instanceof ModemInterfaceAddressImpl)) {
+			return false;
+		}
+		ModemInterfaceAddressImpl other = (ModemInterfaceAddressImpl) obj;
```
As above

Commit:
#779: Unit tests for Modem classes from core.net

- Prepared unit tests for classes ModemInterfaceAddressConfigImpl and
  ModemInterfaceConfigImpl from package org.eclipse.kura.core.net.modem
- Fixed or implemented comparison of classes:
  ModemInterfaceAddressConfigImpl and ModemInterfaceAddressImpl

Signed-off-by: Djuro Drljaca <djuro.drljaca@comtrade.com>
